### ISSUE TITLE
UI conflict with Tidy UI

### ISSUE THREAD

**Issue by Weissrolf** [mouse0270/module-credits#14]
Hello.

The new big buttons at the top of the window cause the bottom buttons of the module window to vanish (window needs to be scrolled) when MC is used in combination with Tidy UI.

**Comment by Weissrolf** [comment]
Screenshot:
https://i.imgur.com/H3yk1mR.png

**Comment by mouse0270** [comment]
I will look into patching this but given that both of these modules do the same thing on this screen, my patch is most likely going to just disable TidyUI on the module management window.

As my import export does everything TidyUI does and more, it even supports TidyUI exports when importing. Does that sound like an acceptable patch, or do you have another suggestion? I am curious as I don't want to basically just disable a modules features.

**Comment by Weissrolf** [comment]
Thanks for the quick reply. As long as Tidy UI's settings improvements keep working I can live with that. :)

Maybe consider making the (overly large) extra spacing and upper buttons font size optional? Leads to lots of scrolling and the upper buttons don't seem important (used often) enough to make them take up so much vertical space.

**Comment by Weissrolf** [comment]
"Settings improvements" meaning the "Configure Settings" being collapsible thanks to Tidy UI. Modules being handled by MC is perfect and TUI's buttons being removed from the top is perfect.

Although, I just notice: TidyUI increases the Module Management window to full browser screen height and then keeps the buttons at the bottom fixed (no need to scroll to get to them). I like that very much, because having to do so much scrolling is something I don't like as UI experience (why use a large display to have small fixed sized windows?).

**Comment by mouse0270** [comment]
So I am not sure what you mean by the heading being larger, yes its bigger then default but still smaller the TidyUI's design. I thought I did a good job at hitting a good middle ground. 

![image](https://user-images.githubusercontent.com/564874/152430564-a290f56d-b2f5-4538-a67d-3fcb57b2c216.png)

Now to be fair, my module does show only 15 modules at a time where the default screen is around 20 and TidyUI had a full length view, showing a lot more. Would a good compromise to do something similar to what TidyUI does, and make the module management window the full height? I would actually like that feature and will add that.

I actually plan to redo the "configure settings" window as well, which at that point would make it a choice over my module or TidyUI cause at that point the major features of both will exist in both modules and there would be no reason to support both.

**Comment by mouse0270** [comment]
TL;DR Would a good temporary compromise on the heading space be to just make the module window full height similar to TidyUI?

**Comment by Weissrolf** [comment]
Larger was more about the combination of both modules. These are all minor things anyway. Having the window full height would be nice, because I personally like to have modules accessible as fast as possible if I go to that list. I don't know if others like that, though. And I can just use the Popout module for my personal preference.

Same with the spacing. Personally I even use the Compact Compendium module to get more entries on screen without scrolling. Even more so because all these lists are just headers anyway, not full journals that you read line by line. Cannot make everyone happy (unless there is an option) and the benefits of your module far outweigh the small drawbacks.

Love the new features and made sure that my own module makes good use of your buttons. Incidentally my module also was one of the "author field as array breaks stuff" ones, because I edited authors to author when I created the original manifest.

Not sure if we need the extra URL button instead of just clicking the version button, but both ways are fine.

**Comment by mouse0270** [comment]
I will look into a small patch to make my module and TidyUI work better together until I can add my own settings page. Cause I do agree with you, its a little annoying to have to scroll down cause of a header that my module basically makes obsolete anyways.

As for the spacing, I think this might be a, I use a 4k monitor issue and I enjoy the look of the extra spacing its easier on the eyes. However, I am not opposed to a "compact" setting that will help reduce the spacing and sizing for those who prefer things closer together.

Yeah, the authors field is kinda a headache, I've had 3 people this week reach out to me asking why their module authors don't show up right...

The URL button being seperate was the cause of foundry's recent change to the tags, it broke mine a little bit, so instead of modifying their tags, I just add my one. However, I am pretty sure I am going to merge the URL back into version, cause I kinda hate it personally. I just realized before I pushed out this update that I never added that tag back in for the 1.0.0 release.

**Comment by Weissrolf** [comment]
I think the only time where the spacing even matters is when people go through modules for initial setup or troubleshooting. There is lots to try around and click then.  I use a 2k display, mostly with browser zoom set to 110% in Foundry (with browsers *not* adhering to Windows DPI settings, so it's real 100%).

Currently you are using 6 pixels above the version button and 9 pixels below. Without MC Foundry uses 4+4. Maybe make MC use 6+6 or 7+7 instead to give it a more symmetric appearance and cause just a bit less empty space?

**Comment by mouse0270** [comment]
Assuming you are referring to the content here: 
![image](https://user-images.githubusercontent.com/564874/152442051-274e4710-30f6-4adf-9d0b-e182f95b6035.png)

I am only getting 7 and 8, not 6 and 9? If you are using TidyUI, the size might be different as it applies it stylings along with mine. As for the 1 pixel difference for me I assume its an issue with `flex` not being able to `justify-content: center` at a `0.5` value.

**Comment by Weissrolf** [comment]
Ah, yes, sorry. Without other modules it is 7+8, with the same total of 15. Foundry uses 4+6 and Tidy UI uses 4+4.

**Comment by mouse0270** [comment]
What about this: 
![image](https://user-images.githubusercontent.com/564874/152443761-167ec590-ef5f-4b2d-b1a2-a952d9926cb6.png)
5+5 for 10...

**Comment by Weissrolf** [comment]
Looks good, but only if this isn't too compressed for your own preference. The right ^ button looks a bit squished between the divider lines now. Maybe that button should shrink?

I also just notice that the text in version buttons is aligned towards the top of the box, while the content in the other buttons is more vertically centered.

**Comment by Weissrolf** [comment]
Last nitpick, the uppermost button row is 35 px and the content slightly misaligned. The + and disk symbol are 11+9 with the dark lines starting at 12 even. The second row below is 34 px and the bottom buttons (Save Module settings) are also 34 px with pseudo 3D border and 30 px without.

So maybe the 1st row can lose a pixel on top just for symmetric perfectionism while we are already counting pixels.  ;)

Thanks again for the quick changes and being open for discussion. I do have a bit unhealthy relationship with free space and scrolling in modern UIs. Antivirus UIs are the worst, large text and icons in fixed sized windows that need lots of scrolling while I sit in front of a 30" display.

**Comment by mouse0270** [comment]
I think any smaller then that and it would bother me. 
As for the sizing on the tags. I think I can fix it... But version specifically will be interesting because its actually the font awesome icon that is making the text look so weirdly placed. So I think for version I will have to add padding to the left to account for the size of the icon, then absolutely position it inside of that space. Its a little weird, but I can fix that too.

Any other bugs I can work on? I love the feedback and want to keep improving the module

**Comment by mouse0270** [comment]
Don't worry about being nitpicky, I spent the last week working on getting the export/import functionality working correctly and just wanted to push the update. I am a little OCD about UI/UX design, so I am all for cleaning up these per pixel issues. 

Trust me in a week when I wouldn't be working on anything, I would be look at all of these issue and they would start eating at the back of my head. lol

**Comment by Weissrolf** [comment]
The mouseover tooltip of the author(s) button looks a bit weird saying "author authors-1".

Other than that I cannot think of anything at this moment. Once I find time I will give the new presets feature a test-ride. I had another module for this installed once, but it broke with v9 (or maybe wasn't reliable to begin with).

**Comment by mouse0270** [comment]
Yeah, thats just jank code to be honest. the title isn't meant to be a title, its literally just a list of classes and I did something stupid with the authors tags. For one author, it shows one person, for two it shows two and 3+ it shows 3 people.

I need to localize that and it will fix that, but for now, jank code. lol

**Comment by Weissrolf** [comment]
Then go and publish! ;)

**Comment by mouse0270** [comment]
Is this both more minimal for you and also big enough for you all at once?!?!?! lol
![image](https://user-images.githubusercontent.com/564874/152536813-ce0a05c3-7c55-45f2-9ffa-f4730a12067c.png)

This will be out in update a little later today.
- Made the header inputs a max of 30px tall reducing the header size by at least 10 pixels in tall.
- Reduced the list size from 35px to 25px and fixed weird formatting of tags
- Increased button footer size to 50px, to make it easier to see and click.

**Comment by Weissrolf** [comment]
Yes, looks good to me. 35 modules on your screen (Tidy UI on my screen displays 43). I will happily test it.

I assume the "taskbar" at the lower edge is some kind of Foundry module, not the OS taskbar that the module window now hangs over?

**Comment by mouse0270** [comment]
You might want to recount that (Also thats at 2k cause you said you were using 2k yesterday). I count 39 modules on screen and have 40 with TidyUI... Almost the same amount. lol

Yeah its Taskbar by Ripper with some UI/UX work I did.

**Comment by mouse0270** [comment]
Side by Side
![image](https://user-images.githubusercontent.com/564874/152538897-750453a1-7104-4e66-b9f7-ca7423b1bf01.png)

**Comment by Weissrolf** [comment]
Looks good except for a small graphical glitch with the last ^ arrow button (lower right) on top of the double line.
https://i.imgur.com/6llfDsB.png

**Comment by mouse0270** [comment]
Its where the row just perfectly meets the footer border of the bottoms

**Comment by mouse0270** [comment]
@Weissrolf Figured I'd show you this before hand to see if I can get your feedback?
![image](https://user-images.githubusercontent.com/564874/152582211-82413993-2b0e-4f0b-a83f-e8c819a21f50.png)

**Comment by Weissrolf** [comment]
Looks very nice. Also more pleasant buttons compared to the up arrow buttons in the module configuration.

Having them on the right side might lead to easier misclicks, though, because the button is so far away from the name, even with the rows being slightly colored. Or can the whole row be clicked to expand the settings (which would safe us from aiming at the button)?

**Comment by mouse0270** [comment]
... Those buttons may or may not match these ones now...

Yeah, but button is just there, the whole row is clickable.

**Comment by dmrickey** [comment]
I like the alternating row colors. And I don't care if the icons are on the right or the left. At least with Tidy UI, the whole row is clickable for it to be collapsible/uncollapsible, so as long as it still does that the icon positioning is insignificant

**Comment by mouse0270** [comment]
@dmrickey Its actually a setting called `Big Picture Mode`
![image](https://user-images.githubusercontent.com/564874/152602406-98ae881a-9579-4bac-a0da-c20073b1c39e.png)

**Comment by mouse0270** [comment]
All of this has been included in the latest release 1.1.5
